include/fmt/ranges.h

```diff
@@ -410,31 +412,76 @@ struct range_formatter<
   FMT_CONSTEXPR auto parse(ParseContext& ctx) -> decltype(ctx.begin()) {
     auto it = ctx.begin();
     auto end = ctx.end();
+    detail::maybe_set_debug_format(underlying_, true);
+    if (it == end) {
+      return underlying_.parse(ctx);
+    }
 
-    if (it != end && *it == 'n') {
+    switch (detail::to_ascii(*it)) {
+    case 'n':
+      set_brackets({}, {});
+      ++it;
+      break;
+    case '?':
+      is_debug = true;
       set_brackets({}, {});
       ++it;
+      FMT_FALLTHROUGH;
+    case 's':
+      if (it == end || *it != 's' || !std::is_same<T, Char>::value) {
```

Comment:
I think `it == end || *it != 's'` should go just before `FMT_FALLTHROUGH`.

Reply:
Done, thanks